### PR TITLE
ui-select improvements

### PR DESCRIPTION
**Pull request by christianmemije**
## Changes
* Improved keyboard support
* Added quick match, aka type to highlight.
* Support disabled options
* Move multi-select check boxes to left.

## Seeking feedback on
* Highlighting and keyboard behavior for single and multi-select 

## Notes
* The examples I added in the docs are not finalized, just for quick testing.
* Did not update dist files.

## To test
* `yarn; yarn dev`

### CODE REVIEW

File: src/UiSelect.vue
Comment (christianmemije) [inline src/UiSelect.vue]: Cleaner way for this conditional function call?

File: src/UiSelect.vue
Comment (christianmemije) [inline src/UiSelect.vue:241]: Instead of tracking indexes, I now track the actual option. The caveat is that if there is more than one option with the same value, they will both be marked as selected.

File: docs/index.html
Comment (indirectlylit) [inline docs/index.html]: was this intentional?

File: src/UiSelect.vue
Comment (indirectlylit) [reply src/UiSelect.vue]: make a new method like 

```javascript
handleOptionMouseover(option) {
  if (this.showDropdown) {
    this.highlightOption(option, { autoScroll: false });
  }
}
```

File: docs/index.html
Comment (christianmemije) [reply docs/index.html]: Yes on the dev server this css was unnecessarily being bundled in causing styles to be overwritten.

File: src/UiSelect.vue
Comment (christianmemije) [reply src/UiSelect.vue]: Yup, was trying to go for a one-liner.

**Comment by indirectlylit** [comment]
When it has keyboard focus, the multi-select menu seems to pop up for all keys, even special keys and modifiers. For example, pressing 'alt', 'cmd', 'shift', 'tab', or even 'escape' show the menu.

There's also some weird behavior with 'tab' - if you press it multiple times, it seems to pop up multiple menus:

gifmakerproject26

File: docs/index.html
Comment (indirectlylit) [reply docs/index.html]: maybe just delete this line then?

**Comment by indirectlylit** [comment]
two issues with the type-ahead:

1. No matter what I type - as long as it starts with an 'R' - I always get 'Read'  selected:

    image

2. typing doesn't work at all if I open the menu by clicking with the mouse and then typing

**Review by indirectlylit — CHANGES_REQUESTED** [review]
Looking very good - but a few issues to sort out

**Comment by christianmemije** [comment]
@indirectlylit Addressed issues, ready for another review.

**Comment by jonboiser** [comment]
@christianmemije  Could you into ESC behavior for multi-select? On single-select, ESC closes the dropdown. On multi-select, the dropdown is not closed, but keyboard loses focus on the ui-select.

Also for the "basic" example. The commonality is they have the search bar.

Multiple selection without search seems fine (multiple selection/favorite colors example).

**Comment by christianmemije** [comment]
@jonboiser Note that @indirectlylit was not able to reproduce the `esc` issue you encountered on his Mac.

**Comment by christianmemije** [comment]
@indirectlylit Can I get a final review? Do not merge though, I will merge it after I update the docs and squash the commits.

**Pull request merged**
